x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/autoscaling/NativeMemoryCapacity.java

```diff
@@ -48,61 +69,151 @@ public NativeMemoryCapacity(long tierMlNativeMemoryRequirement, long nodeMlNativ
      * @return a new instance with the merged capacity values
      */
     NativeMemoryCapacity merge(final NativeMemoryCapacity nativeMemoryCapacity) {
-        if (this == nativeMemoryCapacity) return this;
-        long tier = this.tierMlNativeMemoryRequirement + nativeMemoryCapacity.tierMlNativeMemoryRequirement;
-        long node = Math.max(nativeMemoryCapacity.nodeMlNativeMemoryRequirement, this.nodeMlNativeMemoryRequirement);
-        // If the new node size is bigger, we have no way of knowing if the JVM size would stay the same
-        // So null out
```

Comment:
I think this was another bug in the old code.  If either node or tier size is different (higher or lower) then the JVM size could change after scaling.